bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/types/ThermostatModeGsonTypeAdapter.java

```diff
@@ -27,17 +30,22 @@
  *
  * @author Lukas Agethen - Initial contribution
  */
+@NonNullByDefault
 public class ThermostatModeGsonTypeAdapter implements JsonDeserializer<ThermostatMode>, JsonSerializer<ThermostatMode> {
     @Override
-    public ThermostatMode deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
-            throws JsonParseException {
-        String s = json.getAsString();
-        return s == null ? ThermostatMode.UNKNOWN : ThermostatMode.fromString(s);
+    public ThermostatMode deserialize(@Nullable JsonElement json, @Nullable Type typeOfT,
+            @Nullable JsonDeserializationContext context) throws JsonParseException {
+        JsonElement jsonLocal = json;
+        if (jsonLocal != null) {
+            String s = jsonLocal.getAsString();
+            return s == null ? ThermostatMode.UNKNOWN : ThermostatMode.fromString(s);
+        }
+        return ThermostatMode.UNKNOWN;
     }
 
     @Override
-    public JsonElement serialize(ThermostatMode src, Type typeOfSrc, JsonSerializationContext context)
-            throws JsonParseException {
+    public JsonElement serialize(ThermostatMode src, @Nullable Type typeOfSrc,
+            @Nullable JsonSerializationContext context) throws JsonParseException {
         return new JsonPrimitive(src != ThermostatMode.UNKNOWN ? src.getDeconzValue() : null);
```

Comment:
Are you sure that passing `null` to `JsonPrimitive` is valid? Looking at https://github.com/google/gson/blob/master/gson/src/main/java/com/google/gson/JsonPrimitive.java#L60 it checks null and throws an NPE if the value is `null`.

Reply:
You're right, thanks. I changed it to JsonNull.